excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/AuthorisationParser.java

```diff
@@ -25,7 +23,11 @@ default void parseUserRole(final Authorisation entity,
         final String userRole = definition.getString(ColumnName.USER_ROLE);
         final String caseType = definition.getString(ColumnName.CASE_TYPE_ID);
 
-        parseContext.getRole(caseType, userRole).ifPresent(roleEntity -> entity.setUserRole((UserRoleEntity) roleEntity));
+        entity.setUserRoleId(userRole);
+        parseContext.getRole(caseType, userRole).ifPresent(roleEntity -> entity.setUserRole(roleEntity));
+        if (entity.getUserRole() == null) {
+            parseContext.addMissingUserRole(userRole);
+        }
```

Comment:
I'd refactor to:
```
Optional<UserRoleEntity> roleEntity = parseContext.getRole(caseType, userRole);
        if(roleEntity.isPresent()) {
            entity.setUserRole(roleEntity.get());
        } else {
            parseContext.addMissingUserRole(userRole);
        }
```

Reply:
and perhaps extract that block into a private method with a descriptive name like: `setUserRole`

Reply:
Will do the change

Reply:
we can't create private method because AuthorisationParser is interface.. we can create default method but it will expose that to other classes